sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -66,4 +82,39 @@ impl BlobClient {
         let blob_properties: BlobProperties = response.headers().get()?;
         Ok(blob_properties)
     }
+
+    pub async fn download_blob(
+        &self,
+        options: Option<BlobBlobClientDownloadOptions<'_>>,
+    ) -> Result<Response> {
+        let response = self
+            .client
+            .get_blob_blob_client(self.container_name.clone(), self.blob_name.clone())
+            .download(options)
+            .await?;
+        Ok(response)
+    }
+
+    // For now, this is single-shot, block blob hot path only.
+    pub async fn upload_blob(
+        &self,
+        data: RequestContent<Bytes>,
+        overwrite: bool,
+        content_length: i64,
```

Comment:
They can specify a negative content_length? Note that this isn't .NET that doesn't like unsigned types in public APIs. It's totally idiomatic - and quite frequent (probably more than signed types, in my experience) - to specify unsigned integers.

Reply:
Remember: a common mantra of Rust is "if it compiles, it works". Being able to specify a negative numbers - which probably results in an HTTP 400, I imagine - goes against that principal.

Reply:
Good catch, this was definitely an oversight and I will change it to unsigned